src/analysis/souffle/examples/sql_examples/sql_aux.dl

```diff
@@ -0,0 +1,16 @@
+#include "taint.dl"
+#include "fact_test_helper.dl"
+
+.decl confidentialTag(tag: Tag)
+confidentialTag(tag) :- isTag(tag), substr(tag, 0, strlen("conf_")) = "conf_".
+
+allTests(cat("topQueryNotConfidential_", tag)) :- confidentialTag(tag).
+testPasses(cat("topQueryNotConfidential_", tag)) :- confidentialTag(tag), (!mayHaveTag("top.exprResult", tag)).
+
+.decl doesXform(ap: AccessPath, xform: symbol)
+.decl hasIntegTag(ap: AccessPath, tag: Tag)
+.decl hasValue(ap: AccessPath, value: symbol)
+.decl isIntegTag(tag: Tag)
+isIntegTag(tag) :- isTag(tag), substr(tag, 0, strlen("integ_")) = "integ_".
+hasIntegTag(ap, tag) :- hasTag(ap, tag), isIntegTag(tag).
+hasIntegTag(tgt, tag) :- isIntegTag(tag), resolvedEdge(src, tgt), hasIntegTag(src, tag), !doesXform(tgt, _).
```

Comment:
For integrity, shouldn't all `src` have `tag` for `tgt` to also have `tag`?

Reply:
You are correct. I was focused on calculations clearing the integrity tag and forgot that they must also come in on all input paths.

Reply:
Fixed this.

Reply:
Nope, my fix was dumb. It ended up being an "exists" check rather than a "forall" check. I've got another idea on how to do this.